### PR TITLE
CS Audit, Lime 174, Improved documentation

### PR DESCRIPTION
**Pull request by saxenism**
## Description

In AaveYield.sol
• The specification for the `unlockShares` refers to the asset as the underlying token, but the code assumes to be the Aave token. 

• In _withdrawETH the comment lock collateral is incorrect.

## Change Log

Erroneous documentation were corrected.

## Function Signature Change

The function signature for `function unlockShares` was changed from 

```
function unlockShares(address asset, uint256 amount) external returns (uint256 received);
```

to

```
function unlockShares(address asset, uint256 amount) external returns (uint256 received, address strategyTokens);
```

### CODE REVIEW

File: contracts/yield/AaveYield.sol [hunk 1]
```diff
@@ -231,7 +231,7 @@ contract AaveYield is IYield, Initializable, OwnableUpgradeable, ReentrancyGuard
 
     /**
      * @notice Used to unlock shares
-     * @param asset the address of underlying token
+     * @param asset the address of underlying token, which in this would always be Aave tokens
```
Comment (prateekreddy) [inline contracts/yield/AaveYield.sol:234]: Let's instead use underlying asset only here and change all the instances where this is used

Comment (prateekreddy) [reply contracts/yield/AaveYield.sol:234]: It will save gas costs as we are making multiple calls to yield which can be avoided

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/yield/AaveYield.sol (hunk 1 above)
Comment (saxenism) [reply contracts/yield/AaveYield.sol]: This is done. Please check again.

File: contracts/yield/AaveYield.sol [hunk 2]
```diff
@@ -245,20 +245,23 @@ contract AaveYield is IYield, Initializable, OwnableUpgradeable, ReentrancyGuard
...
+     * @param asset the address of token locked
      * @param amount the amount of shares to unlock
      * @return received amount of shares received
+     * @return address of Aave shares of asset 
      **/
-    function unlockShares(address asset, uint256 amount) external override onlySavingsAccount nonReentrant returns (uint256) {
+    function unlockShares(address asset, uint256 amount) external override onlySavingsAccount nonReentrant returns (uint256, address) {
         if (amount == 0) {
-            return 0;
+            return (0, address(0));
```
Comment (prateekreddy) [inline contracts/yield/AaveYield.sol:255]: Let's still return the correct liquidityToken for correctness

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/yield/AaveYield.sol (hunk 2 above)
Comment (saxenism) [reply contracts/yield/AaveYield.sol]: Yes, correct. Changed the return value. 

Please review again.

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**